src/ConstructionBase.jl

```diff
@@ -150,12 +157,24 @@ function setproperties_namedtuple(obj, patch)
     check_patch_properties_exist(res, obj, obj, patch)
     res
 end
+@generated function check_patch_fields_exist(obj, patch)
+    fnames = fieldnames(obj)
+    pnames = fieldnames(patch)
+    for pname in pnames
+        if !in(pname, fnames)
+            msg = """
+            Failed to assign fields $pnames to object with fields $fnames.
+            """
+            return :(throw(ArgumentError($msg)))
+        end
+    end
+    :(nothing)
+end
 function check_patch_properties_exist(
     nt_new::NamedTuple{fields}, nt_old::NamedTuple{fields}, obj, patch) where {fields}
     nothing
 end
 @noinline function check_patch_properties_exist(nt_new, nt_old, obj, patch)
```

Comment:
`setproperties_namedtuple` uses it, which still uses the `merge(obj, patch)` functionality, but I guess it can be handled by the generated function just the same and therefore all `setproperties_namedtuple` can be removed?

Reply:
Removed in https://github.com/JuliaObjects/ConstructionBase.jl/pull/91/commits/d13e05f05daf7de990fd78e6619cee15e41179f0, seems fine.